not compatible with typescript 4.8.4

when having typescript 4.8.4 getting following errors:
```
../../node_modules/contentful-management/dist/typings/contentful-management.d.ts(52,41): error TS2313: Type parameter 'T' has a circular constraint.
../../node_modules/contentful-management/dist/typings/contentful-management.d.ts(52,92): error TS2536: Type 'number' cannot be used to index type '{ [K in keyof T]: { [x: string]: "workflows"; }; }'.
../../node_modules/contentful-management/dist/typings/contentful-management.d.ts(54,15): error TS2313: Type parameter 'P' has a circular constraint.
```
however it is all fine with typescript 4.2.4

Same issue.

```
$ npx tsc --version
Version 4.8.4

package.json:    "contentful-management": "^10.16.0",

node_modules/contentful-management/dist/typings/contentful-management.d.ts:52:92 - error TS2536: Type 'number' cannot be used to index type '{ [K in keyof T]: { [x: string]: "workflows"; }; }'.

 52 declare function createClient<T extends (ReadonlyArray<string> | readonly ['workflows']) & {
                                                                                               ~
 53     [K in keyof T]: {
    ~~~~~~~~~~~~~~~~~~~~~
...
 55     };
    ~~~~~~
 56 }[number]>(params: ClientOptions, opts: {
    ~~~~~~~~~

node_modules/contentful-management/dist/typings/contentful-management.d.ts:54:15 - error TS2313: Type parameter 'P' has a circular constraint.

54         [P in K]: 'workflows';
```

--- a/lib/contentful-management.ts
+++ b/lib/contentful-management.ts
@@ -62,9 +62,9 @@ function createClient(
   }
 ): PlainClientAPI
 function createClient<
-  // T describes an array that contains the string "workflows" at any point in the list.
-  T extends (ReadonlyArray<string> | readonly ['workflows']) &
-    { [K in keyof T]: { [P in K]: 'workflows' } }[number]
+  // Originally, we used a magic expression from here: https://stackoverflow.com/a/57958451/4816930
+  // Right now it's fixed to this specific array until there is a need to revisit this.
+  T extends ['workflows']
 >(
   params: ClientOptions,
   opts: {

